runtime/parser2/parser.go

```diff
@@ -264,7 +264,9 @@ func (p *parser) skipSpaceAndComments(skipNewlines bool) (containsNewline bool)
 
 func (p *parser) startBuffering() {
 	p.buffering = true
-	p.bufferedTokens = append(p.bufferedTokens, p.current)
+	if p.bufferPos >= len(p.bufferedTokens) {
```

Comment:
```suggestion
 
 	// starting buffering should only buffer the current token if there’s nothing 
 	// to be read from the buffer, otherwise, the current token would be
 	// buffered twice
	if p.bufferPos >= len(p.bufferedTokens) {
```